src/compas_rhino/geometry/surfaces/surface.py

```diff
@@ -163,57 +186,64 @@ def from_rhino(cls, rhino_surface):
         :class:`compas_rhino.geometry.RhinoSurface`
 
         """
-        curve = cls()
-        curve.rhino_surface = rhino_surface
-        return curve
+        from warnings import warn
+
+        warn("RhinoSurface.from_rhino will be removed in the future. Use RhinoSurface.from_native instead.", DeprecationWarning)
+        return cls.from_native(rhino_surface)
 
     @classmethod
-    def from_plane(cls, plane, box):
-        """Construct a surface from a plane.
+    def from_native(cls, native_surface):
+        """Construct a surface from an existing Rhino surface.
 
         Parameters
         ----------
-        plane : :class:`compas.geometry.Plane`
-            The plane.
+        native_surface : :rhino:`Rhino.Geometry.Surface`
+            A Rhino surface.
 
         Returns
         -------
         :class:`compas_rhino.geometry.RhinoSurface`
 
         """
-        plane = plane_to_rhino(plane)
-        box = Rhino.Geometry.BoundingBox(box.xmin, box.ymin, box.zmin, box.xmax, box.ymax, box.zmax)
-        rhino_surface = Rhino.Geometry.PlaneSurface.CreateThroughBox(plane, box)
-        return cls.from_rhino(rhino_surface)
+        instance = cls()
+        instance._rhino_surface = native_surface
+        instance.frame = instance._get_frame_from_planesurface()
+        instance._domain_u = native_surface.Domain(0)[0], native_surface.Domain(0)[1]
+        instance._domain_v = native_surface.Domain(1)[0], native_surface.Domain(1)[1]
```

Comment:
this doesn't make sense to me

Reply:
the domain part also?

Reply:
afaik the u and v domain are read from the native surface by the corresponding properties. they are never set...

Reply:
but my comment was mostly about the frame thing indeed